src/include/ikarus/solver/nonLinearSolver/newtonRaphsonWithScalarSubsidiaryFunction.hh

```diff
@@ -0,0 +1,211 @@
+
+    explicit NewtonRaphsonWithSubsidiaryFunction(
+        const NonLinearOperatorImpl& p_nonLinearOperator,
+        LinearSolver&& p_linearSolver = [](const typename NonLinearOperatorImpl::ValueType& a,
+                                           const typename NonLinearOperatorImpl::ValueType& b) { return a / b; },
+        std::function<void(ResultType&, const UpdateType&)> p_updateFunction =
+            [](ResultType& a, const UpdateType& b) {
+              using Ikarus::operator+=;
+              a += b;
+            })
+        : nonLinearOperator_{p_nonLinearOperator},
+          linearSolver{std::move(p_linearSolver)},
+          updateFunction{p_updateFunction} {}
+
+    using NonLinearOperator = NonLinearOperatorImpl;
+
+    void setup(const NewtonRaphsonWithSubsidiaryFunctionSettings& p_settings) { settings = p_settings; }
+
+    struct NoPredictor {};
+    template <typename SolutionType = NoPredictor, typename SubsidiaryType>
+    requires std::is_same_v<SolutionType, NoPredictor> || std::is_convertible_v<
+        SolutionType, std::remove_cvref_t<typename NonLinearOperatorImpl::ValueType>>
+        SolverInfos solve(SubsidiaryType& subsidiaryFunction, SubsidiaryArgs& subsidiaryArgs) {
+      this->notify(NonLinearSolverMessages::INIT);
+
+      /// Initializations
+      SolverInfos solverInformation;
+      solverInformation.success = true;
+      auto& x                   = nonLinearOperator().firstParameter();  // x = D (Displacements)
+      auto& lambda              = nonLinearOperator().lastParameter();
+
+      /// Determine Fext0
+      /// It is assumed that Fext = Fext0 * lambda such that dRdlambda = Fext0
+      /// Generalization for Fext0 = Fext0(lambda) is not implemented
+
+      auto lambdaDummy = lambda;
+      auto DDummy      = x;
+      x.setZero();
+      lambda = 1.0;
+      nonLinearOperator().template update<0>();
+      const auto Fext0 = (-nonLinearOperator().value()).eval();  // dRdlambda(lambda)
+      lambda           = lambdaDummy;
+      x                = DDummy;
+
+      Eigen::MatrixX<double> residual2d, sol2d;
+      nonLinearOperator().updateAll();
+      const auto& rx = nonLinearOperator().value();
+      const auto& Ax = nonLinearOperator().derivative();
+
+      Eigen::VectorXd deltaD;
+      deltaD.resizeLike(rx);
+      deltaD.setZero();
+      double deltalambda = 0.0;
+
+      subsidiaryArgs.dfdDD.resizeLike(Fext0);
+      subsidiaryArgs.dfdDD.setZero();
+
+      subsidiaryFunction(subsidiaryArgs);
+      auto rNorm = sqrt(rx.dot(rx));
+      decltype(rNorm) dNorm;
+      int iter{0};
+      if constexpr (isLinearSolver) linearSolver.analyzePattern(Ax);
+
+      /// Iterative solving scheme
+      while (rNorm > settings.tol && iter < settings.maxIter) {
+        this->notify(NonLinearSolverMessages::ITERATION_STARTED);
+
+        /// Two-step solving procedure
+        residual2d.resize(rx.rows(), 2);
```

Comment:
can you refactor this block? As follows

```cpp
    sol2d.resize(rx.rows(),Eigen::NoChange);
    residual2d.resize(rx.rows(),Eigen::NoChange);
    residual2d << -rx, Fext0;
```

[Godbolt](https://godbolt.org/z/5Pdov891x)